src/contextAndReducers/AuthProvider.jsx

```diff
@@ -0,0 +1,53 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+
+const AuthContext = createContext();
+const useAuthorization = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  //   reducer function for handling signUp and login
+
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case "SIGN_IN":
+        console.log(action.payload);
+        return {
+          ...state,
+          name: action.payload.firstName,
+          id: action.payload._id,
+        };
+
+      case "LOG_IN":
+        console.log(action.payload);
+        return {
+          ...state,
+          name: action.payload.firstName,
+          id: action.payload._id,
```

Comment:
remove console.log also `signin` and `login` are returning same thing can collab them in one

Reply:
Yeah we can be but Since we are more focused on making functions pure that's why they are kept this way I make every function pure and instead some arguments it don't need dependency on any other function its self sufficient in that way.
So we can do that though.
Thanks for suggestion I will keep this in mind